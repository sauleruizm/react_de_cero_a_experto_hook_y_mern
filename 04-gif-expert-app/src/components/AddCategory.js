import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
    
    const [inputValue, setInputValue] = useState('');
    const handleInputChange = ( e ) => {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length > 2) {
            setCategories(cats => [inputValue, ...cats]);
            setInputValue('');            
        }
        //console.log('submit echo');
    }
    return (
        <form onSubmit={handleSubmit}>
            <h2>Add category</h2>
            <p>{inputValue}</p>
            <input type="text" 
                value={inputValue} 
                onChange= { handleInputChange }/ >
        </form>
    )
    
}

AddCategory.propTypes =  {
    setCategories: PropTypes.func.isRequired
}