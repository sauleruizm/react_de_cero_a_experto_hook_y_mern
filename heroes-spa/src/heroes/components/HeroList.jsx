import { HeroCard } from './HeroCard'

import { getHeroesByPublisher } from '../helpers'
import { useMemo } from 'react'

export const HeroList = ({ publisher }) => {
	const heroes = useMemo( ()=> getHeroesByPublisher(publisher), [ publisher ] )

	return (
		<div className='row rows-cols-1 row-cols-md-3 g-3'>

			{heroes.map((hero) => {
				// console.table(hero)
				return <HeroCard key={hero.id}  {...hero} />
			})}
		</div>
	)
}
