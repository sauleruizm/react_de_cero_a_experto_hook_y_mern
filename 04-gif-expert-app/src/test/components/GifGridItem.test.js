import React from 'react';
import {shallow} from 'enzyme';
import  {GifGridItem} from '../../components/GifGridItem';

describe('Pruebas para <GifGridItem />', () => {
    
    const title = 'Un título';
    const url = 'https://localhost/img'
    const wrapper = shallow( <GifGridItem title={title} url={url} />);


    test('should be show correctly components', () => {
        
        expect( wrapper).toMatchSnapshot();
    });

    test('debe tener un párrafo con el title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    })
    
    test('Debe de tener la imagen igual al url y alt de los props', () => {
        
        const img = wrapper.find('img');
        //console.log(img.props());
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })
    
    test('debe tener animate__fadeIn', () => {
        
        const cssClassExpected = 'animate__fadeIn';
        const div = wrapper.find('div');
        
        expect(div.hasClass(cssClassExpected)).toBe(true);
    })
    
})
