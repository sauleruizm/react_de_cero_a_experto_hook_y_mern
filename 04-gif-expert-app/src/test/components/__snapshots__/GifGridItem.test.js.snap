// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para <GifGridItem /> should be show correctly components 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un título"
    src="https://localhost/img"
  />
  <p>
    Un título
  </p>
</div>
`;
